application-xpoll-api/src/main/java/com/xwiki/xpoll/internal/DefaultXPollManager.java

```diff
@@ -165,17 +175,14 @@ private void updateWinner(XWikiContext context, XWikiDocument doc) throws XWikiE
             throw new XPollException(String.format(MISSING_XPOLL_OBJECT_MESSAGE,
                 doc.getDocumentReference()));
         }
-        List<BaseObject> xpollVotes = doc.getXObjects(XPOLL_VOTES_CLASS_REFERENCE);
 
-        List<String> proposals = xpollObj.getListValue(PROPOSALS);
+        String pollType = xpollObj.getStringValue(XPOLL_TYPE);
 
-        Map<String, Integer> voteCount = getXPollResults(xpollVotes, proposals);
+        Map<String, Integer> voteCount = getXPollResults(doc.getDocumentReference(), pollType);
 
         List<String> currentWinners = findWinner(voteCount);
 
         xpollObj.set(WINNER, String.join(",", currentWinners), context);
-        doc.setAuthorReference(context.getAuthorReference());
-        context.getWiki().saveDocument(doc, "New Vote", context);
```

Comment:
It works without those two lines of code. I'm not sure why. Might it be because of the JavaScript code [here](https://github.com/xwikisas/application-xpoll/blob/cc4ed8446727f50223261f7371e49c6526995f9e/application-xpoll-ui/src/main/resources/XPoll/XPollJSExtension.xml#L133-L152)?